EIPS/eip-7093.md

```diff
@@ -0,0 +1,409 @@
+2. Use NFTs/SBTs as Guardians for their accounts.
+3. Personalize and implement adaptable recovery policies.
+4. Support novel types of Guardians and recovery policies without needing to upgrade their account contracts.
+5. Enable multiple recovery mechanism support, thereby eliminating single points of failure.
+
+This approach enables users to customize recovery policies without the need to change the smart contract of the account itself.
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+
+
+This EIP consists of four key concepts:
+
+- **Identity**: This denotes the representation of a Guardian's identity on the blockchain. It encapsulates traditional on-chain account types such as Externally Owned Accounts (EOA) and Smart Contract Accounts (SCA). More importantly, it extends to include any identity construct capable of producing construct able to be verified on-chain, like signatures and proofs. This could range from Webauthn/Passkey R1 keys to Email DomainKeys Identified Mail (DKIM) signatures, OpenID tokens, Zero-Knowledge Proofs (ZKP), Non-Fungible Tokens (NFTs), SoulBound Tokens (SBTs), and even types yet to be developed. This comprehensive approach ensures a broad, forward-compatible support for various identity types.
+- **GuardianPermissionVerifier**: This component defines how to verify the signature or proof provided by the guardian. Regardless of whether the Guardian's account is on-chain or off-chain, the GuardianPermissionVerifier is invoked during the recovery process of smart contract accounts that incorporate a social recovery system. Its primary role is to confirm the validity of the Guardian's signature or proof, thereby ensuring the authenticity of the Guardian during the recovery process.
+- **RecoveryPolicyVerifier**: This component offers a flexible interface for validating recovery policies. The flexibility stems from allowing account holders or authorized parties to define and store their recovery policies. During the recovery process, the verification logic is implemented by invoking the specific function of the contract instance adopting this interface. Thus, a wide array of customizable social recovery scenarios can be catered to through different contract instances and policy configurations.
+- **RecoveryAccount**: This component encapsulates the core of the social recovery functionality. It is designed to be flexible, composable, and extensible to adapt to various recovery needs. Each RecoveryAccount is defined by an instance contract, crafted by smart contract developers, which embeds the essential logic for the recovery process.
+
+![social_recovery_flow](../assets/eip-7093/social-recovery-flow.png)
+
+### DataTypes
+
+### `TypesAndDecoders.sol`
```

Comment:
I would strongly recommend against organizing your EIP by file name. Since there will likely be multiple implementations of this standard, they will all organize their code differently. How code is organized is _not_ part of a good EIP, since it doesn't affect the observable properties of an implementation.